modules/ismp/clients/parachain/client/src/lib.rs

```diff
@@ -116,8 +116,16 @@ pub mod pallet {
 				data.consensus_state_id, PARACHAIN_CONSENSUS_ID,
 				"Only parachain consensus updates should be passed in the inherents!"
 			);
-			pallet_ismp::Pallet::<T>::handle_messages(vec![Message::Consensus(data)])?;
-			ConsensusUpdated::<T>::put(true);
+
+			// Handling error will prevent this inherent from breaking block production if there's a
+			// reorg and it's no longer valid
+			if let Err(err) =
+				pallet_ismp::Pallet::<T>::handle_messages(vec![Message::Consensus(data)])
+			{
+				log::trace!(target: "ismp", "Parachain inherent consensus update failed {err:?}");
+			} else {
+				ConsensusUpdated::<T>::put(true);
+			}
```

Comment:
not sure i understand the motivation here

Reply:
When I run the inherent locally and there's a reorg, the collator misses block production for that slot because it reinserts the inherent from the reorged block into the new block.
The inherent is no longer valid at that point.

Reply:
I'll paste an example here in a bit.

Reply:
![image](https://github.com/user-attachments/assets/4f95dbc2-a935-43a6-860b-e42e616ee429)

Reply:
Ahh ok